dashboard/src/pages/TreeDetails/TreeDetailsFilterList.tsx

```diff
@@ -22,38 +22,51 @@ const createFlatFilter = (filter: TFilter): string[] => {
 
 const TreeDetailsFilterList = ({
   filter,
-  onFilter,
 }: ITreeDetailsFilterList): JSX.Element => {
   const flatFilter = useMemo(() => createFlatFilter(filter), [filter]);
+  const navigate = useNavigate({ from: '/tree/$treeId' });
 
   const onClickItem = useCallback(
     (flatValue: string, _: number) => {
       const [field, value] = flatValue.split(':');
-      const newFilter = { ...filter };
-      newFilter[field as TFilterKeys][value] = false;
-      onFilter(newFilter);
+
+      const reallyNewFilter = JSON.parse(JSON.stringify(filter ?? {}));
+      const fieldSection = reallyNewFilter[field as TFilterKeys];
+
+      delete fieldSection[value];
+
+      navigate({
+        search: previousSearch => {
+          return {
+            ...previousSearch,
+            diffFilter: reallyNewFilter,
+          };
+        },
+      });
     },
-    [filter, onFilter],
+    [filter, navigate],
   );
 
   const onClickCleanALl = useCallback(() => {
-    const newFilter = { ...filter };
-
-    flatFilter.forEach(flatValue => {
-      const [field, value] = flatValue.split(':');
-      newFilter[field as TFilterKeys][value] = false;
+    navigate({
+      search: previousSearch => {
+        return {
+          ...previousSearch,
+          diffFilter: {},
+        };
+      },
     });
-
-    onFilter(newFilter);
-  }, [filter, onFilter, flatFilter]);
+  }, [navigate]);
 
   return (
-    <FilterList
-      items={flatFilter}
-      onClickItem={onClickItem}
-      onClickCleanAll={onClickCleanALl}
-      removeOnEmpty={true}
-    />
+    <>
```

Comment:
why adding a fragment here?

Reply:
fixed in
https://github.com/kernelci/dashboard/pull/93/commits/d1fd009e2c70dd340f5e557bc17ea6cf2b971c1d